### PR TITLE
Niloofar / CI implementations 

### PR DESCRIPTION
**Pull request by niloofar-deriv**

### CODE REVIEW

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name: test-workflow
+on: push
```
Comment (ako-deriv) [inline .github/workflows/ci-cd.yml]:
```suggestion
on: pull_request
```

Comment (ako-deriv) [reply .github/workflows/ci-cd.yml]: in order to run it on PR level it should be `on: pull_request`

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+name: test-workflow
+on: pull_request
```
Comment (ako-deriv) [inline .github/workflows/ci-cd.yml]:
```suggestion
on: [pull_request]
```

File: .github/workflows/ci-cd.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+name: test-workflow
+on: [pull_request]
```
Comment (ako-deriv) [inline .github/workflows/ci-cd.yml]:
```suggestion
on:
  pull_request:
    
```

File: .github/workflows/ci-cd.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
```
Comment (ako-deriv) [inline .github/workflows/ci-cd.yml]:
```suggestion
on:
  pull_request:
    branches:
    - main
```

File: .github/workflows/ci-cd.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  pull_request:
+    branches:
+    - main
+    
+jobs:
+  sample-job:
+    runs-on: ubuntu-latest
+    steps:
+      - name: my-step
+        run: echo "Hello World!"
```
Comment (ali-hosseini-deriv) [inline .github/workflows/ci-cd.yml:12]: uses: actions/hello-world-javascript-action@main
            with:
              who-to-greet: 'Mona the Octocat'

Comment (ako-deriv) [inline .github/workflows/ci-cd.yml]:
```suggestion
        uses: actions/hello-world-javascript-action@main
          with:
            who-to-greet: 'Mona the Octocat'```

**Comment by ako-deriv** [comment]
merging it to test if its working or not.

**Pull request merged**